### PR TITLE
Add tooltip for copy address button

### LINKED ISSUE #370: [TASK] add copied tooltip on address from wallet selection view
# Overview
It would be preferential for users to be notified when they have successfully copied their address from the Wallet selection modal. 

When a user selects their wallet and proceeds to copy their address a small tooltip should be displayed indicating the address was copied and then close the modal accordingly. 

**Current Wallet modal:**
Screen Shot 2022-03-23 at 2 13 25 PM

## Solution
Similar to MetaMask where a user scrolls on an address and a tooltip is displayed "Copy to clipboard" and then once selected displays "Copied!". 

Screen Shot 2022-03-23 at 2 15 15 PM

**Copied:**
Screen Shot 2022-03-23 at 2 15 24 PM

### Deliverable

- [x] Add a copied tooltip once the user has copied the address successfully 

### Relevant Files

- You can view the current implementations CopyToClipboard component [here](https://github.com/webb-tools/webb-dapp/blob/4fee717a3ecd548aa5a6573b3da15f29e4c83841/packages/ui-components/src/Inputs/WalletSelect/WalletDetails.tsx#L158). In addition, you can see an example from the npm package used [here](https://www.npmjs.com/package/react-copy-to-clipboard).

### PR DESCRIPTION
**Pull request by AtelyPham**
- feat: add copyable hook for reusable
- feat: add tooltip for copy address button

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

CleanShot 2022-06-02 at 18 27 04@2x

Issue Number: Closes #370 

## What is the new behavior?

CleanShot 2022-06-02 at 18 11 33@2x

Screen Shot 2022-06-02 at 18 12 07

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

Thank you @dutterbutter for the comprehensive description in the [issue](https://github.com/webb-tools/webb-dapp/issues/370) 😇.

### CODE REVIEW

**Comment by AtelyPham** [comment]
I think that would be fine. Let me update it! You want to use only the `copy-to-clipboard` dependency and remove `react-copy-to-clipboard`, right?

**Comment by AtelyPham** [comment]
Filter `NaN` values.

Before             |  After
:-------------------------:|:-------------------------:
CleanShot 2022-06-03 at 00 12 28@2x | CleanShot 2022-06-03 at 00 24 32@2x

File: yarn.lock [hunk 1]
```diff
@@ -13213,7 +13206,7 @@ rc@^1.2.7, rc@^1.2.8:
     minimist "^1.2.0"
     strip-json-comments "~2.0.1"
 
-react-copy-to-clipboard@^5.0.2, react-copy-to-clipboard@^5.0.4:
+react-copy-to-clipboard@^5.0.4:
```
Comment (nepoche) [inline yarn.lock:13209]: Can we remove the unused `@polkadot/react-identicon` dependency as well? That way this react-copy-to-clipboard will be removed from our project.

**Review by nepoche — COMMENTED** [review]
Looks great! Just requesting one more change to dependencies but other than that it is good to merge.

File: packages/react-hooks/src/useCopyable.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import copyToClipboard from 'copy-to-clipboard';
+import { useEffect, useRef, useState } from 'react';
+
+type SetTimeoutReturnType = ReturnType<typeof setTimeout>;
+
+/**
+ *
+ * @param {number} deplay The deplay time to reset time copy state in milisecond (default 3000)
```
Comment (nepoche) [inline packages/react-hooks/src/useCopyable.ts:8]:
```suggestion
 * @param {number} display - The display time to reset time copy state in milliseconds (default 3000)
```

File: packages/react-hooks/src/useCopyable.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ * @returns {Object} An object contains `isCopied`, `copiedText` and `copy` function
+ */
+export const useCopyable = (deplay: number = 3000) => {
```
Comment (nepoche) [inline packages/react-hooks/src/useCopyable.ts:11]:
```suggestion
export const useCopyable = (display: number = 3000) => {
```

File: packages/react-hooks/src/useCopyable.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  const copy = (value: string) => {
+    if (isCopied) {
+      return;
+    }
+
+    ref.current = value;
+    copyToClipboard(value);
+    setIsCopied(true);
+
+    const timeoutObj = setTimeout(() => setIsCopied(false), deplay);
```
Comment (nepoche) [inline packages/react-hooks/src/useCopyable.ts:25]:
```suggestion
    const timeoutObj = setTimeout(() => setIsCopied(false), display);
```

Comment (AtelyPham) [reply packages/react-hooks/src/useCopyable.ts]: Updated

File: packages/react-hooks/src/useCopyable.ts (hunk 3 above)
Comment (AtelyPham) [reply packages/react-hooks/src/useCopyable.ts]: Updated

File: packages/react-hooks/src/useCopyable.ts (hunk 2 above)
Comment (AtelyPham) [reply packages/react-hooks/src/useCopyable.ts]: Updated

**Comment by AtelyPham** [comment]
Updated, thanks for @nepoche 's review 💟 Really appreciated it.

**Review by nepoche — APPROVED** [review]

File: packages/bridge/src/components/DepositConfirm/DepositConfirm.tsx [hunk 5]
```diff
@@ -174,19 +183,18 @@ export const DepositConfirm: React.FC<DepositInfoProps> = ({
...
-              <Tooltip title={`Copy note the clipboard`}>
-                <CopyToClipboard onCopy={handleCopy} text={note} {...({ className: 'copy-button' } as any)}>
-                  <IconButton>
+              <Actions>
+                <Tooltip title={'Download Note'}>
+                  <IconButton className={'download-button'} onClick={downloadNote}>
+                    <Icon>download</Icon>
+                  </IconButton>
+                </Tooltip>
+                <Tooltip title={isCopied ? `Copied` : `Copy note the clipboard`}>
```
Comment (dutterbutter) [inline packages/bridge/src/components/DepositConfirm/DepositConfirm.tsx:192]: `Copy to clipboard`

**Review by dutterbutter — CHANGES_REQUESTED** [review]

File: packages/mixer/src/components/DepositConfirm/DepositConfirm.tsx [hunk 6]
```diff
@@ -190,19 +199,18 @@ export const DepositConfirm: React.FC<DepositInfoProps> = ({ mixerSize, onClose,
...
-              <Tooltip title={`Copy note the clipboard`}>
-                <CopyToClipboard onCopy={handleCopy} text={note} {...({ className: 'copy-button' } as any)}>
-                  <IconButton>
+              <Actions>
+                <Tooltip title={'Download Note'}>
+                  <IconButton className={'download-button'} onClick={downloadNote}>
+                    <Icon>download</Icon>
+                  </IconButton>
+                </Tooltip>
+                <Tooltip title={isCopied ? `Copied` : `Copy note the clipboard`}>
```
Comment (dutterbutter) [inline packages/mixer/src/components/DepositConfirm/DepositConfirm.tsx:208]: `Copy to clipboard`

**Pull request merged**